AmplifyPlugins/API/Sources/AWSAPIPlugin/AppSyncRealTimeClient/AppSyncRealTimeRequestAuth.swift

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+import Foundation
+
+public enum AppSyncRealTimeRequestAuth {
+    case cognito(CognitoAuth)
+    case apiKey(ApiKey)
+    case iam(IAM)
```

Comment:
are we missing AWS_Lambda auth or OIDC here? Or is OIDC auth using `.cognito(CogniotAuth)`? technically Cognito should just be treated as an OIDC provider, as that's what it is

Reply:
yeah, as `AWS_Lambda`, `OIDC` and `CognitoUserPools` are all providing the way to get the authToken, they provids the same auth payload. So I [unified them](https://github.com/aws-amplify/amplify-swift/pull/3527/files#diff-8b7d0ea3a610253777061a06962e1d7bc43b6d3a266892b8102d6c053ccc4bc4R94-R118) at auth payload level. Do you think that is reasonable?